django_emarsys/apps.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import re
+
+from django.apps import AppConfig
+from django.core.checks import register
+
+from django_emarsys.config import validate_settings
+
+
+class DjangoEmarsysConfig(AppConfig):
+    name = 'django_emarsys'
+    verbose_name = "Django Emarsys"
+
+    def ready(self):
+        @register()
+        def check_settings(app_configs):
+            messages = validate_settings()
+            for message in messages:
+                # apparently Django's check framework can't handle
+                # unicode correctly, so replace all non-ascii characters
+                # with '?'
+                message.msg = re.sub(r'[^\x00-\x7F]+', '?', message.msg)
```

Comment:
`.encode('ascii', 'replace')` is simpler.

But I don't buy that django's check framework can't handle unicode correctly.

I suspect that that's a python 2 issue which is convinced stdout can only understand ASCII. This can be worked around like so: https://github.com/machtfit/machtfit-oscar/blob/43d1fc7af86db67aaa1afee61f2edc1721f18ee9/apps/catalogue/management/commands/generate_products.py#L42

At which point is there actually a character in the message that's not in ASCII?